lib/node_modules/@stdlib/math/generics/statistics/incrprod/lib/incrprod.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var frexp = require( '@stdlib/math/base/special/frexp' );
+var ldexp = require( '@stdlib/math/base/special/ldexp' );
+
+/**
+* Returns an accumulator function which incrementally computes a product.
+*
+* @returns {Function} accumulator function
+*
+* @example
+* var accumulator = incrprod();
+*/
+function incrprod() {
+	var product = 1.0;
+	var productFactorized = frexp( product );
+
+	return accumulator;
+
+	/**
+	* If provided a value, the accumulator function returns an updated product. If not provided a value, the accumulator function returns the current product.
+	*
+	* @param {number} [x] - new value
+	* @returns {number} product
+	*
+	* @example
+	* var product = accumulator();
+	* // returns 1.0
+	*
+	* product = accumulator( 2.0 );
+	* // returns 2.0
+	*
+	* product = accumulator( -5.0 );
+	* // returns -10.0
+	*
+	* product = accumulator();
+	* // returns -10.0
+	*/
+	function accumulator( x ) {
+		var newFactorized = frexp( x );
+
+		if ( arguments.length === 0 ) {
+			return product;
+		}
+
+		// Multiply mantissa
+		productFactorized[ 0 ] *= newFactorized[ 0 ];
```

Comment:
What is the advantage of keeping the fraction and exponent stored in an `array`? May be a bit more clear to at `L17` or thereabout

``` javascript
var exponent;
var fraction;

// Initial product is 1.0, which may be split into its fractional and exponent parts (0.5 x 2.0**1 = 1.0):
fraction = 0.5;
exponent = 1.0;
```
Then at `L40`,

``` javascript
var parts;
if ( arguments.length === 0 ) {
    return ldexp( fraction, exponent );
}
// Split an incoming value into its fractional and exponent parts:
parts = frexp( x );

// Update the fraction:
fraction *= parts[ 0 ];

// Update the accumulated exponent:
exponent += parts[ 1 ];

// Ensure that the fraction is normalized to avoid overflow/underflow...
if ( fraction < 0.5 && fraction > -0.5 ) {
    // ...
}
// Return a double:
return ldexp( fraction, exponent );
```